drivers/gpio/gpio_kb1200.c

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) ENE
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#define DT_DRV_COMPAT ene_kb1200_gpio
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <soc.h>
+
+/* GPIO module instances */
+#define KB1200_GPIO_DEV(inst) DEVICE_DT_INST_GET(inst),
+static const struct device *const gpio_devs[] = {
+	DT_INST_FOREACH_STATUS_OKAY(KB1200_GPIO_DEV)
+};
+/* Platform specific GPIO functions */
+const struct device *kb1200_get_gpio_dev(int port)
+{
+	if (port >= ARRAY_SIZE(gpio_devs)) {
+		return NULL;
+	}
+
+	return gpio_devs[port];
+}
+
+#define GPIO_REG_BASE       ((GPIO_T*)DT_REG_ADDR_BY_NAME(DT_NODELABEL(gpio0x1x), gpio1x))
+#define GPTD_REG_BASE       ((GPTD_T*)DT_REG_ADDR_BY_NAME(DT_NODELABEL(gpio0x1x), gptd1x))
+
+struct gpio_kb1200_data {
+    /* gpio_driver_data needs to be first */
```

Comment:
Please format all these new files with the [.clang-format](https://github.com/zephyrproject-rtos/zephyr/blob/main/.clang-format) file provided at the root of the repo, Zephyr uses `tab` instead of `space` for alignment